cpp/CMakeLists.txt

```diff
@@ -242,6 +242,7 @@ add_library(cugraph SHARED
     src/traversal/nvgraph_sssp.cu
     src/traversal/two_hop_neighbors.cu
     src/snmg/blas/spmv.cu
+    src/connect/connectivity.cu
```

Comment:
@afender Are we following NetworkX in directory structures? If yes, NetworkX has strongly_connected.py and weakly_connected.py under components (networkx/algorithms/components).

Reply:
Yes. Good catch.

Reply:
So, what exactly is the action item for me here? And ...why do we have to follow networkx dir structure?

Reply:
> So, what exactly is the action item for me here? 
The recommendation is to rename directory `connect` into `components`.

> Why do we have to follow networkx dir structure?
I'd say it is preferred but not required. They already did a lot of categorization work and we often look at Nx API for reference comparison. Using the same directories helps to navigate between both libraries for developers. If an Nx directory doesn't make sense we shouldn't reuse it obviously but I don't think this is the case here.

Reply:
action item: src/connect => src/components
why: I cannot come with very strong reason for this, but AFAIK, this is the basic assumption in previous directory restructuring (so unless there is strong reason to not follow this, it might be better not to break previous assumption). And we are updating cugraph's public API to follow NetworkX's convention, and following NetworkX's directory structure may (or may not) make this task slightly easier...

Reply:
Also, when did we decide to follow networkx dir structure? I might've missed the memo.

Reply:
I'm sorry, why do we want to mimic a certain package (be it NetworkX) or another? I understand providing a certain API conformity with a given package (although even that is debatable). But _mimic_ their directory structure? Aren't we then sending the message that our library is in some way subservient to NetworkX? What if/when something else comes along that we need to provide a similar API for?

Reply:
```
Aren't we then sending the message that our library is in some way subservient to NetworkX? 
```

This is more like the relationship between Pandas and cuDF, cuDF faithfully follows Pandas API as much as possible not because cuDF is inferior to Pandas, but Pandas has a large number of users and we want to promote cuDF as the GPU acceleration library for Pandas. The same logic applies here, and we want to promote cuGraph as an acceleration library for NetworkX. We want to make switching from NetworkX to cuGraph as easy possible (Ideally, just replacing import networkx as nx with import cugraph as nx though we may not get there in short term). This does not mean cuGraph is inferior to NetworkX.

Whether there is a really huge benefit in following NetworkX's directory structure is debatable, but whether we follow or not, the difference in practice is not really that big. And as we already restructured cuGraph's directory structure to follow NetworkX, I don't think there is really enough value in debating over whether this is the right decision or not; the difference is minimal, and we'd better not waste our energy on needless debate.

Reply:
Agreed. It is not that there is a very strong reason for using Nx naming but more that there was no reason to spend time debating about new ones. I don't think it sends a message or makes cugraph subservient in any way.

Reply:
Ok, so I'll rename `connect` to `components`, then. That resolves this conversation?

Reply:
There was no compelling reason to follow NetworkX structure.  However there was a compelling reason to change from the single folder that contained everything approach that we were using.  We picked starting with the NetworkX structure since they had already defined the various analytics types.  If you have justification for why a directory should be called something different, or that some analytics should be saved in a different folder, we can discuss that.

Reply:
Yes, but unless I misunderstood your point, @BradReesWork, this wasn't the case here: I've already provided a separate folder. It was just called `connect` rather than what netx calls it. In any case I changed the name to `components`.